subsystems/networking/networkmanager_linux.go

```diff
@@ -861,21 +872,29 @@ func (n *Networking) mainLoop(ctx context.Context) {
 				if n.Config().TurnOnHotspotIfWifiHasNoInternet.Get() {
 					lastConnectivity = n.connState.getLastOnline()
 				}
+			} else {
+				// Did not get what we need in previous provision (e.g. bad wifi password).
+				forceStartPMode = true
```

Comment:
How does this distinguish between the regular (old) case of "tryCandidates failed to connect" and "tryCandidates tried a **newly-configured** network (and failed)?"

It seems like it would just always hit this and set force-start to true, whether or not a user had just submitted a new wifi config.

Reply:
What I'm saying is this triggers force start on EVERY failure, not just on newly added networks. That means it just always ignore the timer.